TypeScript error when using Yarn PnP

### Describe the bug

### Error

The following error may appear while type checking, if using Yarn PnP:
```
yarn tsc             
.yarn/unplugged/vitest-virtual-635245b105/node_modules/vitest/dist/reporters-5f784f42.d.ts:15:25 - error TS2307: Cannot find module 'rollup' or its corresponding type declarations.

15 import * as rollup from 'rollup';
                           ~~~~~~~~

Found 1 error in .yarn/unplugged/vitest-virtual-635245b105/node_modules/vitest/dist/reporters-5f784f42.d.ts:15
```

### Why this happens:

`vitest` imports `rollup`, while not declaring it as its dependency. This is illegal in Yarn PnP, and works "by miracle" in non-PnP setups.

Here's why.

Imagine `vitest` relies on `foo@1.0.0` which in turn relies on `bar`. `foo`'s maintainer finds a way to get rid of `bar` dependency without changing its public API. Releases this improvement as a non-breaking version, righfully so.

Now, if you relied on `foo` to provide `vitest` with `bar`, everything falls apart.

In fact, with Vite plans to replace `rollup` with `rolldown` the above is EXACTLY what might happen.

### What we can do about it

Declare `rollup` as `vitest`'s dependency.


### Reproduction

n/a

### System Info

```shell
System:
    OS: Windows 10 10.0.22621
    CPU: (8) x64 11th Gen Intel(R) Core(TM) i5-1135G7 @ 2.40GHz
    Memory: 5.44 GB / 15.85 GB
  Binaries:
    Node: 20.9.0 - ~\AppData\Local\Temp\xfs-515b5efc\node.CMD
    Yarn: 4.0.0 - ~\AppData\Local\Temp\xfs-515b5efc\yarn.CMD
    npm: 10.1.0 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: Chromium (118.0.2088.76)
```


### Used Package Manager

yarn

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide/).
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

--- a/packages/vite-node/src/types.ts
+++ b/packages/vite-node/src/types.ts
@@ -34,10 +34,10 @@ export interface StartOfSourceMap {
 export type { EncodedSourceMap, DecodedSourceMap, SourceMapInput } from '@jridgewell/trace-mapping'
 
 export interface RawSourceMap extends StartOfSourceMap {
-  version: string
+  version: number
   sources: string[]
   names: string[]
-  sourcesContent?: string[]
+  sourcesContent?: (string | null)[]
   mappings: string
 }
 

--- a/packages/vitest/src/node/workspace.ts
+++ b/packages/vitest/src/node/workspace.ts
@@ -6,6 +6,7 @@ import type { ViteDevServer, InlineConfig as ViteInlineConfig } from 'vite'
 import { ViteNodeRunner } from 'vite-node/client'
 import { ViteNodeServer } from 'vite-node/server'
 import c from 'picocolors'
+import type { RawSourceMap } from 'vite-node'
 import { createBrowserServer } from '../integrations/browser/server'
 import type { ArgumentsType, Reporter, ResolvedConfig, UserConfig, UserWorkspaceConfig, Vitest } from '../types'
 import { deepMerge } from '../utils'
@@ -140,12 +141,12 @@ export class WorkspaceProject {
       || this.browser?.moduleGraph.getModuleById(id)
   }
 
-  getSourceMapModuleById(id: string) {
+  getSourceMapModuleById(id: string): RawSourceMap | null | undefined {
     const mod = this.server.moduleGraph.getModuleById(id)
     return mod?.ssrTransformResult?.map || mod?.transformResult?.map
   }
 
-  getBrowserSourceMapModuleById(id: string) {
+  getBrowserSourceMapModuleById(id: string): RawSourceMap | null | undefined {
     return this.browser?.moduleGraph.getModuleById(id)?.transformResult?.map
   }
 

